server-spi/src/main/java/org/keycloak/models/CredentialValidationOutput.java

```diff
@@ -63,6 +67,26 @@ public CredentialValidationOutput merge(CredentialValidationOutput that) {
     }
 
     public enum Status {
-        AUTHENTICATED, FAILED, CONTINUE
```

Comment:
Not a blocker as it is related to naming. I think we can make the statues a bit more clear if:

* `VALID` (AUTHENTICATED)
* `INVALID` (FAILED)
* `VALID_USER` (FALLBACK)
* `CHALLENGE`

My point is:

*  `AUTHENTICATED`: Credentials are validated and authentication happens on the upper layer where the authentication logic is in
* `FAILED`, the validator is actually returning that the credential is invalid
* `VALID_USER`, the validator is actually returning if the user is valid but not the credential. The decision to try other providers to authenticate the user depends on the upper layer.
* `CONTINUE`, this name indicates for me that validation should continue and not necessarily the authentication (somewhat related to the `FALLBACK` which indicates moving forward with the provider chain.

Reply:
Thanks Pedro for the suggestions. Your naming is probably better, but there is also concern with backwards compatibility as `CredentialValidationOutput` class is part of the `server-spi` module, which is part of user-storage SPI and hence backwards compatibility might be a concern...

@ahus1 What do you think? Can we rename things or is it better to keep as it is?

I see options like:
1) Rename as @pedroigor  suggested
2) Rename only `FALLBACK` to `VALID_USER` . This is fine for backwards compatibility though as `FALLBACK` is new thing added in this PR
3) Keep as it is

I vote for 2. I can do 1 if @ahus1 agrees that backwards compatibility is not an issue.

Reply:
Update: In the end I've kept `FALLBACK` instead of `VALID_USER`. The `VALID_USER` is not accurate as it is used even for the cases when the user-storage provider is not able to recognize the credential nor the user and hence it just needs to fallback to next provider.

I see options like:
1) Rename statuses to `VALID`, `INVALID`, `FALLBACK`, `CHALLENGE`
2) Keep statuses as is in this PR 

My vote is 2 just due the backwards compatibility. If we can break backwards compatibility, I am fine with (1).

@pedroigor @ahus1 What do you think?

Reply:
Looking at the docs, I see that this change is targeted for KC22, so it should be backward compatible. Therefore I vote to not rename the options.

Reply:
Not a blocker for me.